programs/clearing_house/src/math/amm_jit.rs

```diff
@@ -0,0 +1,123 @@
+use crate::error::ClearingHouseResult;
+use crate::math::orders::standardize_base_asset_amount;
+use crate::math_error;
+use crate::state::market::Market;
+use solana_program::msg;
+
+// assumption: market.amm.amm_jit_is_active() == true  
+// assumption: taker_baa will improve market balance (see orders.rs & amm_wants_to_make)
+pub fn calculate_jit_base_asset_amount(
+    market: &Market,
+    taker_base_asset_amount: u128,
+) -> ClearingHouseResult<u128> {
+    // simple impl
+    // todo: dynamic on imbalance
+    let mut jit_base_asset_amount = standardize_base_asset_amount(
+        taker_base_asset_amount
+            .checked_div(2)
+            .ok_or_else(math_error!())?,
+        market.amm.base_asset_amount_step_size,
+    )?;
+
+    if jit_base_asset_amount > 0 {
+        jit_base_asset_amount =
```

Comment:
this could end up being unstandardized, right? maybe wait to standardize till the end of this function?